### PR TITLE
Add com.tercad.fingrom

### PR DESCRIPTION
**Pull request by lyskouski**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
    - [x] Issue: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found 
    - [x] ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Exception: Couldn't find the executable zenity in the path.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. https://github.com/lyskouski/app-finance
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

**Comment by lyskouski** [comment]
- [ ] Issue: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found

File: com.tercad.fingrom.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+command: fingrom
+separate-locales: false
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --share=network
+  - --share=ipc
+  - --device=dri
+  - --own-name=com.tercad.fingrom
```
Comment (hfiguiere) [inline com.tercad.fingrom.yml:14]:
```suggestion
```

it's granted by default

File: com.tercad.fingrom.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+modules:
+  - name: Fingrom
+    buildsystem: simple
+    only-arches:
+      - x86_64
```
Comment (hfiguiere) [inline com.tercad.fingrom.yml:19]: this doesn't prevent building an aarch64 package.

File: com.tercad.fingrom.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+app-id: com.tercad.fingrom
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
```
Comment (hfiguiere) [inline com.tercad.fingrom.yml:3]: 23.08 is the current runtime

File: com.tercad.fingrom.yml (hunk 2 above)
Comment (hfiguiere) [reply com.tercad.fingrom.yml:19]: why not building from source? This would solve you other issue iwth GLIBC

File: com.tercad.fingrom.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        <li>Shortcuts</li>
+    </ul>
+  </description>
+  <launchable type="desktop-id">com.tercad.fingrom.desktop</launchable>
+  <releases>
+    <release version="5.0.0" date="2023-10-30"/>
+  </releases>
+  <content_rating type="oars-1.1">
+  </content_rating>
+  <update_contact>support@tercad.pl</update_contact>
```
Comment (hfiguiere) [inline com.tercad.fingrom.metainfo.xml:43]: missing mandatory screenshots.

File: com.tercad.fingrom.svg
Comment (hfiguiere) [inline com.tercad.fingrom.svg:1]: doesn't upstream have an icon?

File: com.tercad.fingrom.desktop
Comment (hfiguiere) [inline com.tercad.fingrom.desktop:1]: Doesn't upstream show a desktop file?

File: com.tercad.fingrom.desktop [hunk 5]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Version=1.0.0.0
```
Comment (hfiguiere) [inline com.tercad.fingrom.desktop:2]: this is an invalid version entry.

File: com.tercad.fingrom.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.tercad.fingrom</id>
+  <name>Fingrom</name>
+  <summary>Open-source platform-agnostic financial accounting application</summary>
```
Comment (hfiguiere) [inline com.tercad.fingrom.metainfo.xml:5]: please don't lie in the summary. `CC-BY-NC-ND-4.0` is NOT open source. This package might even be breaking the license since it it a derivative work.

File: com.tercad.fingrom.yml (hunk 2 above)
Comment (hfiguiere) [reply com.tercad.fingrom.yml:19]: I see, Flutter....

File: com.tercad.fingrom.metainfo.xml (hunk 6 above)
Comment (hfiguiere) [reply com.tercad.fingrom.metainfo.xml:5]: nonwithstanding that Creative Commons licenses are not code licenses.

File: com.tercad.fingrom.svg
Comment (lyskouski) [reply com.tercad.fingrom.svg:1]: It has, but in other formats (png, ico); for the next release might shift it back to the source code.

File: com.tercad.fingrom.desktop
Comment (lyskouski) [reply com.tercad.fingrom.desktop:1]: For the next release will shift it back to the source code.

File: com.tercad.fingrom.metainfo.xml (hunk 6 above)
Comment (lyskouski) [reply com.tercad.fingrom.metainfo.xml:5]: Someone except the author is not eligible to distribute. What's wrong with that?

Comment (lyskouski) [reply com.tercad.fingrom.metainfo.xml:5]: `Creative Commons Licenses: by making your data publicly available, you retain ownership of your objects`

File: com.tercad.fingrom.yml (hunk 2 above)
Comment (lyskouski) [reply com.tercad.fingrom.yml:18]: > this doesn't prevent building an aarch64 package.
As I see from https://github.com/flathub/im.fluffychat.Fluffychat/blob/master/im.fluffychat.Fluffychat.json it should work.

File: com.tercad.fingrom.metainfo.xml (hunk 6 above)
Comment (lyskouski) [reply com.tercad.fingrom.metainfo.xml:10]: Surely, it's not a free license, content used "with author permission".

> That makes it source available not open source.

Ok, let me rephrase then, if you're arguing to "redistributed" as inevitable part of the project.

> And both the things you said about the CC-BY-NC-ND-4.0 license are wrong....

What exactly?

CC-BY-NC-ND or Creative Commons Attribution NonCommercial NoDerivs. With this license, the user (while attributing the original creator) can only share the work but not change it in any way or ever use it commercially.
This is technically still considered open since it does not have a commercial charge for sharing the work, but it does not offer any flexibility with the material should someone wish to remix and adapt it.

File: com.tercad.fingrom.yml (hunk 2 above)
Comment (bbhtt) [reply com.tercad.fingrom.yml:18]: You missed https://github.com/flathub/im.fluffychat.Fluffychat/blob/master/flathub.json#L2 and the docs https://docs.flathub.org/docs/for-app-authors/maintenance#dont-build-on-aarch64 ...

File: com.tercad.fingrom.svg
Comment (bbhtt) [reply com.tercad.fingrom.svg:1]: If the png is 128px or greater you can add use that one instead of svg.

File: com.tercad.fingrom.metainfo.xml (hunk 6 above)
Comment (bbhtt) [reply com.tercad.fingrom.metainfo.xml:10]: > Open source = sources of the project are freely accessible. No contradictions

That makes it source available not open source.

And both the things you said about the CC-BY-NC-ND-4.0 license are wrong.

> Someone except the author is not eligible to distribute, or forking without pushing back the made change.

Comment (bbhtt) [reply com.tercad.fingrom.metainfo.xml:10]: In fact it will show up as proprietary in software stores/website as AppStream.license_is_free_license returns false.

<details> 
<summary> Image </summary>

5b8344f1b6

</details>

File: com.tercad.fingrom.yml (hunk 2 above)
Comment (lyskouski) [reply com.tercad.fingrom.yml:18]: How it differs from https://github.com/flathub/flathub/pull/4668/files#diff-449c727070caefacc6c03cb10922640e75b1ef6f2540973d535734f063aeb5f5R17  (lines 17-18)?

Comment (bbhtt) [reply com.tercad.fingrom.yml:18]: You don't have a flathub.json that is necessary...

Comment (lyskouski) [reply com.tercad.fingrom.yml:18]: Thanks, added

File: com.tercad.fingrom.metainfo.xml (hunk 6 above)
Comment (bbhtt) [reply com.tercad.fingrom.metainfo.xml]: > Ok, let me rephrase then, if you're arguing to "redistributed modified versions" as inevitable part of the open-source project.

Yes that is an essential part to qualify for open source, see the definition anywhere. You also should be using a OSI/FSF approved license.

Creative Commons licenses are explicitly not software license https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software

If you want to use a license that restricts those freedoms, feel free but call it "Source code is available", not "open source". That term is only applied when certain conditions like above are met.

File: flathub.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+    "only-arches": [
+      "x86_64"
+    ],
+    "automerge-flathubbot-prs": true
```
Comment (bbhtt) [inline flathub.json:5]: This is a bad idea, you already ran into one error using precompiled binaries. Builds will get pushed to users without any testing

Comment (lyskouski) [reply flathub.json:5]: It's going to be a part of an automated distribution. 

As for the errors, existing one gonna to be solved. Ongoing will fail the build and prevent the package from its distribution.

File: com.tercad.fingrom.metainfo.xml (hunk 6 above)
Comment (lyskouski) [reply com.tercad.fingrom.metainfo.xml]: I do understand that it's not recommended; but not restricted. Possibly, I've missed the one with defined conditions as for CC-BY-NC-ND-4.0. Furthermore, that project is a part of the book, and the book is a part of that project - https://www.amazon.com/Zero-Market-Flutter-Desktop-Distribution-ebook/dp/B0CLLKNXQR . They affect each other and use the single repository.

**Comment by bbhtt** [comment]
bot, build com.tercad.fingrom

File: com.tercad.fingrom.yml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      - install -Dm644 com.tercad.fingrom.metainfo.xml /app/share/appdata/com.tercad.fingrom.metainfo.xml
+    sources:
+      - type: file
+        url: https://github.com/lyskouski/app-finance/releases/download/v5.0.1/fingrom_Linux.tar.gz
+        # sha256sum fingrom_Linux.tar.gz
+        sha256: ce16ac7c02914f9fa845b8da1d5f2656a4d9e23d33c1c73c78c2af5d22931a1f
+      - type: file
+        path: com.tercad.fingrom.svg
+      - type: file
+        path: com.tercad.fingrom.desktop
```
Comment (bbhtt) [inline com.tercad.fingrom.yml:37]:
```suggestion
```

File: com.tercad.fingrom.yml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      - x86_64
+    build-commands:
+      - mkdir -p Fingrom
+      - tar -xf fingrom_Linux.tar.gz -C Fingrom
+      - cp -r Fingrom /app/Fingrom
+      - chmod +x /app/Fingrom/fingrom
+      - mkdir -p /app/bin
+      - ln -s /app/Fingrom/fingrom /app/bin/fingrom
+      - mkdir -p /app/lib
+      - ln -s /usr/lib/x86_64-linux-gnu/libstdc++.so.6 /app/lib/libstdc++.so.6
```
Comment (bbhtt) [inline com.tercad.fingrom.yml:28]: This does not seem necessary.

**Comment by bbhtt** [comment]
bot, build com.tercad.fingrom

File: com.tercad.fingrom.yml (hunk 9 above)
Comment (lyskouski) [reply com.tercad.fingrom.yml]: You're right. Dropped

File: com.tercad.fingrom.yml [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+command: fingrom
+separate-locales: false
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --share=network
+  - --share=ipc
+  - --device=dri
+  - --filesystem=home
```
Comment (bbhtt) [inline com.tercad.fingrom.yml:14]: Why do you need full home access?

Comment (lyskouski) [reply com.tercad.fingrom.yml:14]: Only to ".terCAD"-folder to store applications' data.

Comment (bbhtt) [reply com.tercad.fingrom.yml:14]: Add `--persist=.terCAD` and remove it. This will create the folder relative to real home inside the per app sandbox folder. Ideally you should follow the xdg standard and use XDG_* locations to store app data.

Comment (lyskouski) [reply com.tercad.fingrom.yml:14]: Hm.. not related to that access. Updated main thread.

Comment (lyskouski) [reply com.tercad.fingrom.yml]: Resolved. Removed `--filesystem=home`, `persist` is not needed as well. Thanks

File: flathub.json (hunk 7 above)
Comment (lyskouski) [reply flathub.json]: Got it, replaced by `false`

**Comment by lyskouski** [comment]
> What has been changed in the latest commit?

```
{
    "only-arches": [
      "x86_64"
    ],
-    "automerge-flathubbot-prs": true
+ "automerge-flathubbot-prs": false
  }
```
